__tests__/randomUUID.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const assert = require('assert');
+
+const randomUUID = require('../randomUUID');
+
+
+function testMatch(uuid) {
+  assert.match(
+    uuid,
+    /^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/);
+}
+
+
+// Generate a number of UUID's to make sure we're not just generating the same
+// value over and over and to make sure the batching changes the random bytes.
+test("Generate multiple UUIDs", function()
+{
+  const last = new Set([
+    '00000000-0000-0000-0000-000000000000'
+  ]);
+
+  for (let n = 0; n < 130; n++) {
+    const uuid = randomUUID();
+    assert(!last.has(uuid));
+    last.add(uuid);
+    assert.strictEqual(typeof uuid, 'string');
+    assert.strictEqual(uuid.length, 36);
+    testMatch(uuid);
+
+    // Check that version 4 identifier was populated.
+    assert.strictEqual(
+      Buffer.from(uuid.substr(14, 2), 'hex')[0] & 0x40, 0x40);
```

Comment:
Simpler to just validate the char directly.
```suggestion
    assert.strictEqual(uuid.substr(14, 1), '4');
```

Reply:
The test is extracted from Node.js tests too.